File: backend/src/models/pet.ts
import mongoose from "mongoose";


const petSchema = new mongoose.Schema({
  _id: { type: String, required: true },
  name: { type: String, required: true },
  age: { type: String, required: true },
  weigh: { type: String, required: true },
  breed_id: { type: String, ref: 'BreedType', required: true },
  email: { type: String, ref: 'Owner', required: true },
  sex: { type: String, required: true },
  breed_type: { type: String, required: true },
  img: { type: String, required: true },
  medic_id: {type: [String], ref: 'Medic', required: false},
  record_id : {type: [String], ref: 'Record', required: false}
});

const Pet = mongoose.model("pet", petSchema, "pet");
export default Pet;

File: flutter_petcare_app/lib/features/auth/data/models/pet.dart


class Pet {
  final String id;
  final String name;
  final String age;
  final String weigh;
  final String breedId;
  final String ownerId;
  final String sex;
  final String breedType;
  final String img;

  Pet({
    required this.id,
    required this.name,
    required this.age,
    required this.weigh,
    required this.breedId,
    required this.ownerId,
    required this.sex,
    required this.breedType,
    required this.img,
  });

  factory Pet.fromJson(Map<String, dynamic> json) {
    return Pet(
      id: json['_id'],
      name: json['name'],
      age: json['age'],
      weigh: json['weigh'],
      breedId: json['breed_id'],
      ownerId: json['owner_id'],
      sex: json['sex'],
      breedType: json['breed_type'],
      img: json['img'],
    );
  }
}